tracee-ebpf/tracee/events_processor.go

```diff
@@ -216,16 +216,16 @@ func (t *Tracee) processEvent(event *external.Event) error {
 			return err
 		}
 	case SchedProcessExitEventID:
-		delete(t.processTree.processTreeMap, event.HostProcessID)
+		delete(t.processTree.processTreeMap, event.HostThreadID)
 	case SchedProcessForkEventID:
-		hostTid, _ := getEventArgUint32Val(event, "child_tid")
-		childPid, _ := getEventArgUint32Val(event, "child_pid")
-		pid, _ := getEventArgUint32Val(event, "child_ns_pid")
-		ppid, _ := getEventArgUint32Val(event, "parent_ns_pid")
-		hostPpid, _ := getEventArgUint32Val(event, "parent_pid")
-		childTid, _ := getEventArgUint32Val(event, "child_ns_tid")
-		processData := ProcessCtx{event.Timestamp, event.ContainerID, pid, childPid, ppid, hostTid, hostTid, hostPpid, uint32(event.UserID), uint32(event.MountNS), uint32(event.ProcessID)}
-		t.processTree.processTreeMap[int(childTid)] = processData
+		hostTid, _ := getEventArgInt32Val(event, "child_tid")
+		hostPid, _ := getEventArgInt32Val(event, "child_pid")
+		pid, _ := getEventArgInt32Val(event, "child_ns_pid")
+		ppid, _ := getEventArgInt32Val(event, "parent_ns_pid")
+		hostPpid, _ := getEventArgInt32Val(event, "parent_pid")
+		tid, _ := getEventArgInt32Val(event, "child_ns_tid")
+		processData := ProcessCtx{event.Timestamp, event.ContainerID, uint32(pid), uint32(tid), uint32(ppid), uint32(hostTid), uint32(hostPid), uint32(hostPpid), uint32(event.UserID), uint32(event.MountNS), uint32(event.PIDNS)}
```

Comment:
Although the original type of the argument is an int, there is no meaning to get the int32 value, then convert to uint32, as the size of the variable is 32 bits in both (so if theoretically we had a negative value, the conversion won't take it into account). so, you can just use getEventArgUint32Val like you did before and avoid this extra conversion

Reply:
I agree, but I talked with @roikol and he says that getEventArgUint32Val return's error to him because it is defined as int

Reply:
the `getEventArgUint32Val()` func tries to assing the value as uint32: `val, ok := arg.Value.(uint32)`. it fails here because it is of type int32.
what we can do is use the new func intruduced here, but return a converted value: return `uint32(val)` or `return int(val)`.